src/rust/ide/ast/impl/src/lib.rs

```diff
@@ -231,12 +231,18 @@ impl Ast {
         self
     }
 
-    /// Wraps given shape with an optional ID into Ast.
+    /// Wraps given shape with ID into Ast (uses random new ID if None).
```

Comment:
I'd like it much more if this was split to two separate functions: `new` that takes only `Shape` and the other one (like new_with_id) that takes Shape and non-optional Id.